server.go

```diff
@@ -183,13 +204,16 @@ func parseURLParameters(url, pattern string) map[string]string {
 		return nil
 	}
 
-	parameters := make(map[string]string)
+	parameters := make(map[string]string, len(patternSegments))
```

Comment:
The len argument is optional. It helps the runtime to initialize the map with a good size.